tiledb/sm/query/query_condition.cc

```diff
@@ -1671,14 +1678,20 @@ void QueryCondition::apply_ast_node_sparse(
   Datatype type = Datatype::UINT64;
   std::string node_field_name = node->get_field_name();
   if (node_field_name != constants::timestamps) {
-    const auto attribute = array_schema.attribute(node_field_name);
-    if (!attribute) {
-      throw std::runtime_error("Unknown attribute " + node_field_name);
-    }
+    if (array_schema.is_dim(node_field_name)) {
```

Comment:
@KiterLuc I think this section can use the arrayschema getters as well (unless I'm missing something)